test/UrlParserTest.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright 2024, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Julian Mundhahs (mundhahj@tf.uni-freiburg.de)
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "util/GTestHelpers.h"
+#include "util/http/HttpUtils.h"
+#include "util/http/UrlParser.h"
+
+using namespace ad_utility;
+
+TEST(UrlParserTest, paramsToMap) {
+  auto parseParams =
+      [](const string& queryString) -> HashMap<std::string, std::string> {
+    boost::urls::url_view url(queryString);
+    return UrlParser::paramsToMap(url.params());
+  };
+  auto HashMapEq = [](const HashMap<std::string, std::string>& hashMap)
+      -> testing::Matcher<HashMap<std::string, std::string>> {
+    return testing::ContainerEq(hashMap);
+  };
+
+  EXPECT_THROW(parseParams("?foo=a&foo=b"), std::runtime_error);
+  // Parameter with key "" is set twice.
+  EXPECT_THROW(parseParams("?&"), std::runtime_error);
+
+  EXPECT_THAT(parseParams("?query=SELECT%20%2A%20WHERE%20%7B%7D"),
+              HashMapEq({{"query", "SELECT * WHERE {}"}}));
+  EXPECT_THAT(parseParams("?cmd=stats"), HashMapEq({{"cmd", "stats"}}));
+  EXPECT_THAT(parseParams("/ping"), HashMapEq({}));
+  EXPECT_THAT(parseParams(""), HashMapEq({}));
+  EXPECT_THAT(parseParams("?"), HashMapEq({{"", ""}}));
+  EXPECT_THAT(parseParams("/ping?a=b&c=d"),
+              HashMapEq({{"a", "b"}, {"c", "d"}}));
+  EXPECT_THAT(parseParams("?foo=a"), HashMapEq({{"foo", "a"}}));
+  EXPECT_THAT(parseParams("?foo=a"), HashMapEq({{"foo", "a"}}));
+  EXPECT_THAT(parseParams("?a=b&c=d&e=f"),
+              HashMapEq({{"a", "b"}, {"c", "d"}, {"e", "f"}}));
+  EXPECT_THAT(parseParams("?=foo"), HashMapEq({{"", "foo"}}));
+  EXPECT_THAT(parseParams("?=foo&a=b"), HashMapEq({{"", "foo"}, {"a", "b"}}));
+  EXPECT_THAT(parseParams("?foo="), HashMapEq({{"foo", ""}}));
+  EXPECT_THAT(parseParams("?foo=&bar=baz"),
+              HashMapEq({{"foo", ""}, {"bar", "baz"}}));
+}
+
+TEST(UrlParserTest, parseRequestTarget) {
+  auto IsParsedUrl = [](const string& path,
+                        const HashMap<std::string, std::string>& parameters)
+      -> testing::Matcher<const UrlParser::ParsedUrl&> {
+    return testing::AllOf(
+        AD_FIELD(UrlParser::ParsedUrl, ParsedUrl::path_, testing::Eq(path)),
```

Comment:
Does this work?
I though `AD_FIELD(UrlParser::parsedUrl, path_, Eq(path)` was shorter and possibly more correct. But maybe I am confusting something.